### PR TITLE
[WIP] Machinery explosion tweaks

### PR DESCRIPTION
**Pull request by comma**
Now machinery will take more damage to components on higher explosion seveity, and has a chance to dismantle.
Removed a bunch of ex_act overrides in buildable machines, they should be using base one. Adjusted ones with special effects to call parent after they're done.

### CODE REVIEW

File: code/modules/power/lighting.dm [hunk 1]
```diff
@@ -395,15 +395,13 @@
...
 	switch(severity)
-		if(1)
-			qdel(src)
-			return
 		if(2)
 			if (prob(75))
 				broken()
 		if(3)
 			if (prob(50))
 				broken()
```
Comment (afterthought2) [inline code/modules/power/lighting.dm:403]: Doesn't really make sense: at 2 it breaks, but at 1 nothing happens?

File: code/game/machinery/vending/_vending.dm [hunk 2]
```diff
@@ -124,22 +124,9 @@
...
-		if(3.0)
-			if (prob(25))
-				spawn(0)
-					src.malfunction()
-					return
-				return
-		else
-	return
+	if(severity == 3 && prob(25))
+		malfunction()
```
Comment (afterthought2) [inline code/game/machinery/vending/_vending.dm:128]: also here

File: code/modules/power/lighting.dm (hunk 1 above)
Comment (comma) [reply code/modules/power/lighting.dm:403]: They call parent at the end, broken() is lightswitch-specific thing to break bulb

File: code/game/machinery/vending/_vending.dm (hunk 2 above)
Comment (comma) [reply code/game/machinery/vending/_vending.dm:128]: Ditto, it does malfunction check then runs parent which should handle the wholesale destruction.

**Comment by comma** [comment]
As per discorb need to look into if non-constructible machines doing dismantle is okay

**Pull request closed without merging**